mod-source-record-storage-server/src/main/java/org/folio/services/RecordServiceImpl.java

```diff
@@ -126,6 +129,16 @@ public RecordServiceImpl(final RecordDao recordDao) {
     return recordDao.streamSourceRecords(condition, recordType, orderFields, offset, limit, tenantId);
   }
 
+  @Override
+  public Flowable<String> streamMarcRecordIds(String leaderExpression, String fieldsExpression, Boolean deleted, Boolean suppress, int offset, int limit, String tenantId) {
+    if (leaderExpression == null && fieldsExpression == null) {
```

Comment:
Shouldn't this be an or? Currently, either one can be non-null and it would continue. I see the following methods perform a null check as well. Also, should it check for empty string as well? `StringUtils.isEmpty`

Reply:
We throw an exception only when both leaderExpression & fieldsExpression are null. If one of them is not null we just continue the request processing.
There is a validation in place. There are validation rules
https://github.com/folio-org/mod-source-record-storage/pull/313/files#diff-023247c5abe9543eb4d7d877eeb48e32ce1ff7702bfeaa9567e83b6954a22e9c